### PR TITLE
Algolia empty key check

### LINKED ISSUE #57: GitHub actions empty secrets
For some reason GH action has issues accessing the secrets when running the test deploy workflow. The main deploy workflow seems to pick up the secrets fine.

[main deployment link](https://github.com/BUGS-NYU/nyu-cs-wiki/actions/workflows/pages/pages-build-deployment)
[test deployment link](https://github.com/BUGS-NYU/nyu-cs-wiki/actions/workflows/test-deploy.yml)

### PR DESCRIPTION
**Pull request by calvinatian**
Adds a check for empty keys. This is because secrets are not passed to runners from forks as a security concern. This should resolve #57.

### CODE REVIEW

**Review by nedjulius — APPROVED** [review]
looks good. I think we should also add some global variables that determine whether the build is in production or dev, so that we can throw an error if the keys are missing and it's production build as they're prod expected

**Comment by calvinatian** [comment]
Sounds good. I think we can use env variables for that. Since those aren't secret I believe they should be passed to the action.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ jobs:
           ALGOLIA_APP_ID: ${{ secrets.ALGOLIA_APP_ID }}
           ALGOLIA_SEARCH_API_KEY: ${{ secrets.ALGOLIA_SEARCH_API_KEY }}
           ALGOLIA_INDEX_NAME: ${{ secrets.ALGOLIA_INDEX_NAME }}
+          NODE_ENV: ${{ vars.NODE_ENV }}
```
Comment (nedjulius) [inline .github/workflows/deploy.yml:30]: I think it's enough to just do `NODE_ENV: production` here

File: .github/workflows/test-deploy.yml [hunk 2]
```diff
@@ -29,3 +29,4 @@ jobs:
           ALGOLIA_APP_ID: ${{ secrets.ALGOLIA_APP_ID }}
           ALGOLIA_SEARCH_API_KEY: ${{ secrets.ALGOLIA_SEARCH_API_KEY }}
           ALGOLIA_INDEX_NAME: ${{ secrets.ALGOLIA_INDEX_NAME }}
+          NODE_ENV: ${{ vars.NODE_ENV }}
```
Comment (nedjulius) [inline .github/workflows/test-deploy.yml:32]: ... and `NODE_ENV: development` here

File: docusaurus.config.js [hunk 3]
```diff
@@ -12,10 +12,17 @@ function getAlgoliaConfig(
...
   // filter missing required keys
-  const missingKeys = requiredKeys.filter((key) => config[key] === undefined)
+  const missingKeys = requiredKeys.filter(
+    (key) => config[key] === undefined || config[key] === ''
+  )
 
   // if some keys are missing, return undefined
   if (missingKeys.length > 0) {
+    // require all keys in production
+    if (process.env.NODE_ENV == 'production') {
```
Comment (nedjulius) [inline docusaurus.config.js:22]: `process.env.NODE_ENV === 'production'`

Comment (nedjulius) [reply docusaurus.config.js:22]: idk if this is the problem though

Comment (calvinatian) [reply docusaurus.config.js]: I think the issue is `NODE_ENV` is set to 'production' when doing `npm run build`. Should we just use some other variable name?

Comment (nedjulius) [reply docusaurus.config.js]: yeah I think we should prob not overcomplicate this and just change the name

**Pull request merged**